### ISSUE TITLE
Kotlin v.1.2.41 compatilbity

### ISSUE THREAD

**Issue by holgerbrandl** [khud/sparklin#21]
When changing to the latest v1.2.41 release in the build dependencies, it builds fine, but fails when running `bin/kshell.sh` with 
```
[sparklin]$ ./bin/kshell.sh 
Exception in thread "main" java.lang.NoClassDefFoundError: org/jetbrains/kotlin/cli/jvm/compiler/CliLightClassGenerationSupport$NoScopeRecordCliBindingTrace
	at sparklin.kshell.repl.CodeAnalyzer.<init>(CodeAnalyser.kt:33)
	at sparklin.kshell.repl.ReplCompiler.<init>(ReplCompiler.kt:23)
	at sparklin.kshell.KShell.initEngine(KShell.kt:125)
	at sparklin.kshell.KShell.doRun(KShell.kt:139)
	at sparklin.kshell.KotlinShell.main(KotlinShell.kt:32)
Caused by: java.lang.ClassNotFoundException: org.jetbrains.kotlin.cli.jvm.compiler.CliLightClassGenerationSupport$NoScopeRecordCliBindingTrace
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:338)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 5 more

Bye!
```

**Comment by khud** [comment]
Unfortunately compiler classes sometimes move around. I need some to think about it. It is almost impossible to provide compatibility layer between versions (due to some technical issues). I think the best way to solve it is to make `kshell-repl-api` as separate artefact and deploy to Maven Central with the same version as Kotlin compiler.

**Comment by holgerbrandl** [comment]
Having dedicated artifacts on central for supported kotlin versions seems the best solution to me. 

By putting it on maven central you also enable users to write their own custom repl launchers such as 
```
#!/usr/bin/env kscript

@file:DependsOn("org.sparklin:kshell:1.0") // or whatever the locator might be

// user additions to classpath
@file:DependsOn("de.mpicbg.scicomp:krangl:0.9.1")

// launch it
sparklin.kshell.KotlinShell.main(args)
```

**Comment by khud** [comment]
@holgerbrandl I will deploy artifacts to Maven Central soon (I hope it will be tomorrow), but now you can build 1.2.41 - compatible version from 0.2 branch. Checkout and build [kshell-repl-api](https://github.com/khud/kshell-repl-api) from master before.

**Comment by holgerbrandl** [comment]
Thanks for great support!

**Comment by holgerbrandl** [comment]
I've tried to build the v0.2 branch but it fails (on a fresh repo clone). One reason seems that `<module>jline3-shaded</module>` is commented in the parent pom. Also the dependency 
```
    <!--<dependency>-->
            <!--<groupId>com.github.khud</groupId>-->
            <!--<artifactId>kshell-repl-api</artifactId>-->
            <!--<version>0.2.2-1.2.30</version>-->
        <!--</dependency>-->
```
in `kshell/pom.xml` seems invalid to me.

Do I miss something ?

**Comment by holgerbrandl** [comment]
Great fix, thanks @khud. I'm looking forward to the artifacts.

**Comment by holgerbrandl** [comment]
After still struggeling with the compilation, I just noticed that kshell-repl-api has a dedicated repo at https://github.com/khud/kshell-repl-api/tree/0.2-1.2.41 and is not yet on jcenter/central either, so to compile sparklin, the user needs to install it first.

**Comment by khud** [comment]
Fixed in 0.2.3
